### PR TITLE
[unitaryHACK] Added SWAP decomposition and related tests

### LINKED ISSUE #1304: [unitaryhack] Define decomposition for `qml.SWAP`
*This issue has been tagged for contributions during [unitaryHACK](https://unitaryfund.github.io/unitaryhack/participating-projects.html)*

Each PennyLane operation defines a decomposition. The decomposition of an operation is an ordered list of other operations. Applying this list of operations evolves quantum states the same way as applying the operation itself does. In most cases, this list includes elementary gates that are often supported by most quantum devices.

We would like to add the decomposition of the `qml.SWAP` operation by:
1. Defining the decomposition method for [the `qml.SWAP` operation](https://github.com/PennyLaneAI/pennylane/blob/1930230dde60112ad90af0951fa9295d374e1adf/pennylane/ops/qubit.py#L528) that has the following signature:
```python
    @staticmethod
    def decomposition(wires):
```
Other operations in the `pennylane/ops/qubit.py` file can serve as an example for defining this method (e.g., [see `qml.Hadamard`](https://github.com/PennyLaneAI/pennylane/blob/1930230dde60112ad90af0951fa9295d374e1adf/pennylane/ops/qubit.py#L87)).
2. Including a test case for the decomposition of `qml.SWAP` in the `tests/ops/test_qubit_ops.py` file to verify its correctness.

### PR DESCRIPTION
**Pull request by charmerDark**
**Context:**
Added SWAP decomposition and wrote tests to it

**Description of the Change:**
added the the function 
 ``` 
  @staticmethod
    def decomposition(wires): 
```
to ` pennylane/ops/qubit.py `  and tests for it at ` tests/ops/test_qubit_ops.py `

**Possible Drawbacks:**
The tests only run through some specific cases and cannot be considered very conclusive. This also seems to apply to many of the pre-existing tests too
**Related GitHub Issues:**
resolves #1304

### CODE REVIEW

**Review by antalszava — CHANGES_REQUESTED** [review]
Hi @charmerDark, thank you for submitting this PR! :slightly_smiling_face: :+1: Overall, it's starting to look great already!

There would be a couple of adjustments to make:
1. Formatting the files using `black` by running `black -l 100 pennylane/`. The `21.5b1` version of `black` is required for that, it can be installed by `pip install black --upgrade`.
2. a) Adding a one-sentence description and a link to this PR in the `.github/CHANGELOG.md` file. Other PRs could serve as an example there.
    b) Adding yourself as a contributor to the list of contributors! :slightly_smiling_face: 

> The tests only run through some specific cases and cannot be considered very conclusive. This also seems to apply to many of the pre-existing tests too

Could you elaborate on this point from the PR description?

----

On another note, no need to worry about the failing test cases for now. :slightly_smiling_face:  These are due to existing test cases that assumed that the `qml.SWAP` operation has no decomposition (we'll be [adjusting them separately](https://github.com/PennyLaneAI/pennylane/pull/1318)).

**Comment by charmerDark** [comment]
> Could you elaborate on this point from the PR description?
The test written for the SWAP operation assumes that the decomposition is composed of CNOT gates. This cannot be considered a conclusive test in my opinion.

**Comment by charmerDark** [comment]
@antalszava  #1313 is what I was trying to convey in the possible drawbacks section of this PR. I just couldn't articulate it that well

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 <h3>New features since last release</h3>
 
+* The `qml.SWAP`  operation now
+  has decompositions over elementary gates.
```
Comment (antalszava) [inline .github/CHANGELOG.md:6]: (Minor suggestion)
```suggestion
  has a decomposition over elementary gates.
```

**Review by antalszava — CHANGES_REQUESTED** [review]
Hi @charmerdark, thank you for the adjustments! :slightly_smiling_face: Yes, it indeed makes sense to further consider the best way of testing, though it's okay to have that on the side for now.

As for the changes, a couple of small last steps would remain:
1. Thanks for the addition in the changelog! Could you add yourself to the [list of contributors](https://github.com/PennyLaneAI/pennylane/blame/master/.github/CHANGELOG.md#L253)?
2. Formatting the files was great, the check now passes well! :slightly_smiling_face: One thing is, that certain files are good to be left unformatted (e.g., docs, benchmark files, etc.). The command `black -l 100 pennylane/` is meant to be executed from the `pennylane` directory such that it formats the `pennylane` subdirectory where the source files are to be found (this was not specified, sorry if it caused confusion~). Could you perhaps revert the formatting commit using `git` and reformat such that only the `pennylane` subdirectory is formatted?

Let us know if you have any questions! :)

**Comment by charmerDark** [comment]
Wont reverting changes now cause conflict  with the merge from the main that happened right now? I am a bit confused with the `git revert` and how to use it. Could you please help me a bit?

**Comment by antalszava** [comment]
Hi @charmerDark, sure :slightly_smiling_face:

The `git revert 2ff25c789a20156fda3cebb2da92953c70365c14` command should do it, where the string after `revert` is the hash of the formatting commit, 2ff25c789a20156fda3cebb2da92953c70365c14. `git` will apply the "inverse" of the changes in that particular commit when using `git revert`, meaning that it will effectively unchange any updates coming from there. There should be no issues from later merges of `master`. :slightly_smiling_face:

**Comment by charmerDark** [comment]
I made the changes after reverting the branch. However I encountered the error message 
```Updates were rejected because the tip of your current branch is behind its remote counterpart. Integrate the remote changes```
Could you help me with this too?  So sorry you had to babysit me through this.

**Comment by antalszava** [comment]
Hi @charmerDark, it looks the changes were removed, would you happen to have another copy of your changes locally? That way, you could create a new branch and a new pull request. I've also pulled your changes recently, if that would help. Let me know :slightly_smiling_face:

**Comment by charmerDark** [comment]
I'll make a new branch and  another pull request. Sorry this has all turned into such a hassle.

**Comment by antalszava** [comment]
Sure! No worries :slightly_smiling_face: Feel free to tag me and can go for a review.

**Pull request closed without merging**